internal/manifest/v1beta1/parse.go

```diff
@@ -76,3 +87,14 @@ func pullLayer(ctx context.Context, imageRef string, keyChain authn.Keychain) (v
 	}
 	return crane.PullLayer(noSchemeImageRef, crane.WithAuthFromKeychain(keyChain), crane.WithContext(ctx))
 }
+
+// getLockerForPath always returns the same sync.Locker instance for given path argument.
+func getLockerForPath(path string) sync.Locker {
+	val, ok := fileMutexMap.Load(path)
```

Comment:
Why not only use `LoadOrStore`

Reply:
To avoid creating `sync.Mutex{}` objects with every single call, which would not be used - because after some time all the necessary entries in the Map will already have some Mutex assigned.
In this implementation (assume 10 workers), in the worst case, 10 goroutines will invoke this method, 10 `sync.Mutex{}` objects are created, and only one "wins" and becomes a Map entry. The other 9 are garbage collected. But it's just 9 objects instead of one object per every call to the method (essentially an unbounded number).
I know it's not much for GC to "eat" these objects, but if we can easily spare few CPU cycles, why not?

Reply:
tbh, considering readability and clean code purpose, this tiny performance increase is not relevant so much, there are lots of temporary objects created for each reconciling which are left for GC to be cleaned up, you can run with pprof to verify, this additional amount of empty `sync.Mutex{}` are nothing compare to that. I suggest simplifying this.

Reply:
I just realized another problem, the lifetime of this `fileMutexMap` is not permanent, it only lives in each reconciliation loop, which means for each concurrent worker, everytime when it do `fileMutexMap.Load(path)`, they get no result, then all of them will enter into `LoadOrStore`, so basically, the performance gain you imagined also not true.

Reply:
Can you help me to fix the problem you described? How should I change the implementation then so that `fileMutexMap` is permanent?

Reply:
In my opinion, using Load would have a slightly better performance as Tomasz mentioned. I believe the if condition should be met only if the path is not in the map or when a worker is just spawned not with each reconciliation loop right?